pkg/kube/controllers/extendedjob/job_creator.go

```diff
@@ -106,6 +104,45 @@ func (j jobCreatorImpl) Create(ctx context.Context, eJob ejv1.ExtendedJob, names
 		}
 	}
 
+	// Fetch all secrets
+	tokenSecret := corev1.Secret{}
+	secretList := &corev1.SecretList{}
+	err = j.client.List(ctx, secretList, client.InNamespace(namespace))
+	if err != nil {
+		return false, errors.Wrapf(err, "could not get secret list related to ejob %s.", eJob.Name)
+	}
+	for _, secret := range secretList.Items {
+		annotations := secret.GetAnnotations()
+		annotation, ok := annotations["kubernetes.io/service-account.name"]
+		if ok {
+			if annotation == serviceAccountName {
+				tokenSecret = secret
+				break
+			}
+		}
+	}
+
+	// Mount service account token on container
+	secretMode := int32(420)
+	serviceAccountVolumeName := names.Sanitize(fmt.Sprintf("%s-%s", serviceAccount.Name, tokenSecret.Name))
+	serviceAccountVolume := corev1.Volume{
+		Name: serviceAccountVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  tokenSecret.Name,
+				DefaultMode: &secretMode,
+			},
+		},
+	}
+	serviceAccountVolumeMount := corev1.VolumeMount{
+		Name:      serviceAccountVolumeName,
+		ReadOnly:  true,
+		MountPath: "/var/run/secrets/kubernetes.io/serviceaccount",
```

Comment:
Is it guaranteed that in any circumstance, this specified secret will take priority over the pod-mounted service account token?

Reply:
Yes. [0]

[0] https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/serviceaccount/admission.go#L479

Reply:
Perfect.